### ISSUE TITLE
Plankton: test Julabo simulator

### ISSUE THREAD

**Issue by mattclarke** [ISISComputingGroup/IBEX#1796]
I have created a Julabo emulator on a Plankton fork found [here](https://github.com/mattclarke/plankton/tree/master/plankton).
It needs to be tested before it can be added to Plankton/IBEX. It has two protocols to compensate for the different command sets found across models (more than two will be required longer term). The protocol choice can be specified when starting Plankton.

**Testing protocol 1**

From the Plankton directory:
```
python.exe plankton.py -p julabo-version-1 julabo -- --bind-address localhost --port 57677
```

julabo-version-1 should behave like a FP50_MH which is the one set in IBEX, so starting the IOC in simulation mode pointing at port 57677 should work. Confirm that the HIGHLIMIT is 100 and the LOWLIMIT is 0.

**Testing protocol 2**
 
From the Plankton directory:
```
python.exe plankton.py -p julabo-version-2 julabo -- --bind-address localhost --port 57677
```

julabo-version-2 should behave like a FP50-HE (unconfirmed, as the hardware has not be checked). For protocol 2 the commands for HIGHLIMIT and LOWLIMIT are different, and the termination characters are different(!).

The IOC can be modified to allow testing of the new protocol by editing the proto files.

JulaboCommom.proto requires the in-terminator changing to \n:

```
InTerminator = '\n';
```

The JulaboVariable.proto is more complicated as the in-terminator and the high and low limits need changing:

```
InTerminator = '\n';

< other protocols and comments omitted >

readHighLimit {
    out "IN_SP_01";
    in "%f";
    ExtraInput = Ignore;
}

readLowLimit {
    out "IN_SP_02";
    in "%f";
    ExtraInput = Ignore;
}

```

Confirm that the HIGHLIMIT is 100 and the LOWLIMIT is 0 as for protocol 1 and everything else behaves as before.

**Comment by IsabellaRey** [comment]
I have tested the emulator, it mostly works, but a couple of things need fixing:

In device.py:
- [ ] set_set_point() should convert to a double, not to int: for example, int(80.0) throws (and plankton swallows the exception, so you don't see any error message in the plankton shell)
- [ ] set_mode() and get_circulating(): should they be renamed so it's clearer that they are getter/setter of the same thing?
- [ ] set_mode() doesn't work. I have set plankton's StreamAdapter.handle_error() to print out exception messages and I get the following: "set_mode() takes exactly 2 arguments (1 given)"

In device.py and julabo_stream_interface_1.py:
- [ ] our IOC reads the external temperature as IN_PV_02 and the power as IN_PV_01, but they are swapped in interface 1 (though I read in the IOC protocol that they may be swapped depending on the model). Also, the device.py has a self.heating_power field, but it's never used: get_power() returns the self.external_temperature

NOTE: when testing interface 2, I have changed the IOC to use IN_SP_03 for the HighLimit and IN_SP_04 for the LowLimit (i.e. I'm guessing the ticket had it written wrong - these are the numbers used in the interface 2 code), which was fine
